### PR TITLE
made main-navigation-sidebar responsive

### PR DESCRIPTION
**Pull request by hunar1312**
Hello @DonKoko,
The desktop main navigation sidebar is now responsive for all screen sizes as described in the issue #91
Please Review this. Thanks

### CODE REVIEW

File: app/components/layout/sidebar/atoms.ts [hunk 1]
```diff
@@ -1,8 +1,8 @@
 import { atom } from "jotai";
 
-export const sidebarCollapseStatusAtom = atom(false);
```
Comment (DonKoko) [inline app/components/layout/sidebar/atoms.ts]: I had changed those as I prefer the default state to be open.

File: app/components/layout/sidebar/index.tsx [hunk 2]
```diff
@@ -48,68 +46,114 @@ export default function Sidebar({ user }: Props) {
...
-          </span>
         </Link>
-        <button
-          className="transition-all duration-200 ease-linear"
-          onClick={manageUncollapsedSidebar}
-        >
-          <i className="icon text-gray-500">
-            {maintainUncollapsedSidebar ? <ActiveSwitchIcon /> : <SwitchIcon />}
-          </i>
+        <button className="menu-btn relative z-50" onClick={toggleSidebar}>
```
Comment (DonKoko) [inline app/components/layout/sidebar/index.tsx]: Please make a component for this.

File: app/components/layout/sidebar/index.tsx [hunk 3]
```diff
@@ -48,68 +46,114 @@ export default function Sidebar({ user }: Props) {
...
-      <div className="mt-auto">
-        <SidebarBottom
-          user={{
-            ...user,
-          }}
-        />
-      </div>
-    </aside>
+      </header>
+      <div
```
Comment (DonKoko) [inline app/components/layout/sidebar/index.tsx]: This works really good but its super hard to read like this, as its difficult to figure which html is which element. Can you please place this in a separate component.

File: app/components/layout/sidebar/index.tsx [hunk 4]
```diff
@@ -48,68 +46,114 @@ export default function Sidebar({ user }: Props) {
...
+              {maintainUncollapsedSidebar ? (
+                <ActiveSwitchIcon />
+              ) : (
+                <SwitchIcon />
+              )}
+            </i>
+          </button>
+        </div>
+        <div className="flex-1">
+          <ul className="menu mt-6 md:mt-10">
```
Comment (DonKoko) [inline app/components/layout/sidebar/index.tsx]: Can you please make a component MenuItems and put this part of the code there.

File: app/routes/_layout+/_layout.tsx [hunk 5]
```diff
@@ -4,7 +4,7 @@ import type {
   LoaderFunction,
 } from "@remix-run/node";
 import { json } from "@remix-run/node";
-import { Outlet, useLoaderData } from "@remix-run/react";
+import { Link, Outlet, useLoaderData } from "@remix-run/react";
```
Comment (DonKoko) [inline app/routes/_layout+/_layout.tsx]: Here is a linting error.

**Review by DonKoko — CHANGES_REQUESTED** [review]
Overall it looks and works good @hunar1312. I really like it.
Just 1 thing, I see you have used the same atom to control the "keep opened" on desktop and the mobile menu. This works but I would rather use different atoms for it for a few reasons:

1. I want the default state to be 'open' which only makes sense on desktop and if we do it like this, on mobile the menu will be open by default and we dont want that.
2. In the near future when we add some functionality to save this in the user settings, we will again have a conflicts with that.

File: app/components/layout/sidebar/index.tsx [hunk 6]
```diff
@@ -48,68 +46,114 @@ export default function Sidebar({ user }: Props) {
...
+            {menuItems.map((item) => (
+              <li key={item.label}>
+                <NavLink
```
Comment (DonKoko) [inline app/components/layout/sidebar/index.tsx]: Here we need to close the menu on click on mobile. So if the user navigates they actually see the page and dont have to manually close the menu.

File: app/components/layout/sidebar/atoms.ts (hunk 1 above)
Comment (hunar1312) [reply app/components/layout/sidebar/atoms.ts]: Okay wasn't aware of that. Will use different atoms for mobile.

File: app/components/layout/sidebar/index.tsx (hunk 3 above)
Comment (hunar1312) [reply app/components/layout/sidebar/index.tsx]: @DonKoko do you want me to convert this overlay into a component or the whole header?

Comment (DonKoko) [reply app/components/layout/sidebar/index.tsx]: @hunar1312 The overlay

**Review by DonKoko — APPROVED** [review]
Love it! Great job @hunar1312.

**Pull request merged**